misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -187,4 +224,57 @@ private void createReport(final IBundleCoverage bundleCoverage, ModuleCoverage m
         return filteredPathList;
     }
 
+    private void writeReport(final IBundleCoverage bundle) {
+
+        try {
+            final IReportVisitor visitor = createReportVisitor();
+
+            // Not sure what this does. Enable and test
+
+            final ExecFileLoader loader = new ExecFileLoader();
+            File execFile = coverageDir.resolve("ballerina.exec").toFile();
+            loader.load(execFile);
+
+            visitor.visitInfo(loader.getSessionInfoStore().getInfos(), loader.getExecutionDataStore().getContents());
+            visitor.visitBundle(bundle, getSourceLocator());
+            visitor.visitEnd();
+
+        } catch (IOException e) {
+        }
+
+    }
+
+    private IReportVisitor createReportVisitor() throws IOException {
+
+        IReportVisitor visitor = null;
```

Comment:
Please move the declaration to the same line as definition. (its even possible to remove it completely by inlining to return)